src/Module.cpp

```diff
@@ -692,12 +692,7 @@ void Module::compile(const std::map<OutputFileType, std::string> &output_files)
         debug(1) << "Module.compile(): device_code " << output_files.at(OutputFileType::device_code) << "\n";
         Buffer<> buf = get_device_code_buffer();
         if (buf.defined()) {
-            int length = buf.size_in_bytes();
-            while (length > 0 && ((const char *)buf.data())[length - 1] == '\0') {
-                length--;
-            }
-            std::string str((const char *)buf.data(), length);
-            std::string device_code = std::string((const char *)buf.data(), buf.size_in_bytes());
```

Comment:
What am I looking at here? Was `str` just unused?

Reply:
Seems like it was yes. That's a win ! 😁

Reply:
Nice 👍